packages/core/src/rules/arazzo/workflow-dependsOn-unique.ts

```diff
@@ -0,0 +1,56 @@
+import type { ArazzoRule } from '../../visitors';
+import type { UserContext } from '../../walk';
+
+export const WorkflowDependsOnUnique: ArazzoRule = () => {
+  const seenWorkflow = new Set();
+  const existingSourceDescriptions = new Set();
+  const existingWorkflowIds = new Set();
+
+  return {
+    SourceDescriptions: {
+      enter(sourceDescriptions) {
+        for (const sourceDescription of sourceDescriptions) {
+          existingSourceDescriptions.add(sourceDescription.name);
+        }
+      },
+    },
+    Workflows: {
+      enter(workflows) {
+        for (const workflow of workflows) {
+          existingWorkflowIds.add(workflow.workflowId);
+        }
+      },
+    },
+    Workflow: {
+      leave(workflow, { report, location }: UserContext) {
+        if (!workflow.dependsOn) return;
+
+        for (const item of workflow.dependsOn) {
+          // $sourceDescriptions.<name>.<workflowId>
```

Comment:
Have you forgotten to remove the comment?

Reply:
No, just add it here to have some structure awareness

Reply:
Then leave a meaningful comment like this:


```suggestion
          // Checking for this pattern: $sourceDescriptions.<name>.<workflowId>
```